src/types/message.ts

```diff
@@ -0,0 +1,47 @@
+export enum MessageType {
+  Result = "Result",
+  Raw = "Raw"
+}
+
+export enum MessageLevel {
+  Info = "Info",
+  Warning = "Warning",
+  Error = "Error"
+}
+
+export type JsonPath = {
+  tag: string;
+  path: string;
+};
+
+export type Extra = {
+  [key: string]: any;
+};
+
+export type BaseMessageRecord = {
+  level: MessageLevel;
+  message: string;
+  time: Date;
```

Comment:
What's the time used for?  To log execution time?

Reply:
yes